content/kotlin/concepts/functions/functions_in_kotlin.md

```diff
@@ -0,0 +1,194 @@
+```
+fun functionName(argumentName: dataType, argumentName2 : datatype){
+  body
+}
+
+fun sample(num1: Int,num2: Int){
+  body
+}
+```
+When calling the function, `arguments` can be put in their assigned spot or defined by name. By defining by name the `arguments` do not need to be in the same order as in the function header.
+
+```
+fun main(){
+  println(sample(3,5))
+}
+fun main(){
+  println(sample(num2=5, num1=3))
+}
+```
+
+`Arguments` can have default value. It is assigned with equals sign `=`.
+
+```
+fun default(name: String = "Kotlin"){
+  print("Hello $name")
+}
+```
+Default `arguments` can be change when calling a function.
+```
+fun main(){
+  default("Java")
+}
+///output
+///Hello Java
+```
+
+
+### Return type
+
+The `return type` expresses the returning value type. Entering `return type` is optional. `Return type` is after parentheses and colon `:`. By default the return value is `Unit`, `Unit` means there is no returning value.
+
+```
+fun functionName(argument): return type {
+  body
+}
+fun sample(name: String): String {
+  body
+}
+```
+
+### Function body
+
+`Function body` has the instruction for the function. It must be between curly brackets `{}`. `Function body` is indented, except for single line function. The opening curly bracket `{` is at the end of the function header and the closing curly bracket `}` is below the last line of `function body`.
+
+```
+fun functionName(argument) {
+  bodyline1
+  bodyline2
+}
+```
+
+### Return statment
+
+If there is a `return statement` in the function,then there must be also a `return type`. `Return statement` is defined by keyword `return`. Lines after the `return statement` will not be executed.
+
+```
+fun adding(num1: Int,num2: Int): Int{
+  var sum = num1 + num2
+  //Return statement, returns the variable sum
+  return sum
+}
+
+fun adding(num1: Int,num2: Int): Int{
+  var sum = num1 + num2
+  //Return statement, returns the variable sum
+  return sum
+  //println will not execute
+  println("Nice job")
+}
+```
+
+## Single line function
+
+The single line function can be written only if the function has **single expression**.
+
+```
+fun area(side1: Int, side2: Int): Int{
+  return side1 * side2
+}
+```
+Curly brackets `{}` and `return` can be removed and by adding an equals sign `=`, we get the single line function.
+
+```
+fun area(side1: Int, side2: Int): Int = side1 * side2
+```
+
+The compiler can conclude the `return type` also, we can remove the `return type`.
+
+```
+fun area(side1: Int, side2: Int) = side1 * side2
+```
+
+## Function Literals
+
+Function becomes `function literal` when we assign it as a value. There are 2 types of function literals: `Lambda Expression` and `Anonymous Function`.
+
+### Lambda Expression
+
+`Lambda Expression` is a short function between curly braces `{}`. The `return` is replaced by `->`. Adding `return type` is optional because the compiler deduce it from the type interface.
```

Comment:
```suggestion
A lambda expression is a short function between curly braces `{}`. The `return` is replaced by `->`. Adding `return type` is optional because the compiler can infer it from the type interface.
```

Also, this is an incomplete description of the syntax. There's no `fun` keyword, no function name, and no parenthesis.